### PR TITLE
[tensorflow][build] Updated buildspec.yml for tf 2.1.1 and py 3.6

### PR DESCRIPTION
**Pull request by SergTogul**
## Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

*Description:*
Updated buildspec.yml for tf 2.1.1 and py 3.6

*Tests run:*
Executed locally

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by arjkesh** [comment]
@SergTogul are you also planning to build and test the py2 images for your review? If so, you will need to add them back into the buildspec

see https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec-old.yml#L70, but make sure the docker_file path includes cuda path for gpu images 
(i.e. 
```
docker_file: !join [ docker/, *VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
```
)

File: test/dlc_tests/container_tests/bin/testTensorFlow [hunk 1]
```diff
@@ -34,6 +34,7 @@ set -e
 
 echo "Verify tensorflow profiler can generate logs."
 
+
```
Comment (arjkesh) [inline test/dlc_tests/container_tests/bin/testTensorFlow:37]: nit: can you remove, because I think this is causing all images to be built in the PR since it appears change to container_tests directory spawns all builds

**Comment by SergTogul** [comment]
> @SergTogul are you also planning to build and test the py2 images for your review? If so, you will need to add them back into the buildspec
> 
> see https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec-old.yml#L70, but make sure the docker_file path includes cuda path for gpu images
> (i.e.
> 
> ```
> docker_file: !join [ docker/, *VERSION, /, *DOCKER_PYTHON_VERSION, /, *CUDA_VERSION, /Dockerfile., *DEVICE_TYPE ]
> ```
> 
> )

No. Only Python 3.6

File: tensorflow/training/docker/2.1.1/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -108,7 +108,7 @@ RUN mkdir -p /root/.ssh/ \
 WORKDIR /
 
 RUN ${PIP} --no-cache-dir install --upgrade \
-    pip \
+    pip==20.2.1 \
```
Comment (saimidu) [inline tensorflow/training/docker/2.1.1/py3/Dockerfile.cpu:111]: Why is this necessary?

File: tensorflow/training/docker/2.1.1/py3/cu101/Dockerfile.gpu [hunk 3]
```diff
@@ -156,7 +156,7 @@ RUN mkdir -p /root/.ssh/ \
 WORKDIR /
 
 RUN ${PIP} --no-cache-dir install --upgrade \
-    pip \
+    pip==20.2.1 \
```
Comment (saimidu) [inline tensorflow/training/docker/2.1.1/py3/cu101/Dockerfile.gpu:159]: Why is this necessary?

**Review by saimidu — COMMENTED** [review]
Please address the checks on the PR description template, and add more details to the description about the dockerfile changes that have been made.

Also, please update the smdebug version from 0.9.2 to 0.9.3.

File: tensorflow/training/docker/2.1.1/py3/cu101/Dockerfile.gpu (hunk 3 above)
Comment (SergTogul) [reply tensorflow/training/docker/2.1.1/py3/cu101/Dockerfile.gpu:159]: Using pip newer than 20.2.1 leads to a failure of horovod deployment

File: tensorflow/training/docker/2.1.1/py3/Dockerfile.cpu (hunk 2 above)
Comment (SergTogul) [reply tensorflow/training/docker/2.1.1/py3/Dockerfile.cpu:111]: Using pip newer than 20.2.1 leads to a failure of horovod deployment

Comment (saimidu) [reply tensorflow/training/docker/2.1.1/py3/Dockerfile.cpu:111]: nit: If there is another iteration, please add a comment about this to the dockerfile to make it more clear why this was done.

**Review by saimidu — APPROVED** [review]

**Review by arjkesh — APPROVED** [review]
